packages/odie-client/src/data/use-get-unread-conversations.ts

```diff
@@ -0,0 +1,32 @@
+import { HELP_CENTER_STORE } from '@automattic/help-center/src/stores';
+import { useDispatch as useDataStoreDispatch } from '@wordpress/data';
+import Smooch from 'smooch';
+import { ZendeskConversation } from '../types';
+
+const calculateUnread = ( conversations: Conversation[] | ZendeskConversation[] ) => {
+	let unreadConversations = 0;
+	let unreadMessages = 0;
+
+	conversations.forEach( ( conversation ) => {
+		const unreadCount = conversation.participants[ 0 ]?.unreadCount ?? 0;
+
+		if ( unreadCount > 0 ) {
+			unreadConversations++;
+			unreadMessages += unreadCount;
+		}
+	} );
+
+	return { unreadConversations, unreadMessages };
+};
+
+export const useGetUnreadConversations = () => {
+	const { setUnreadCount } = useDataStoreDispatch( HELP_CENTER_STORE );
+
+	return ( conversations?: Conversation[] | ZendeskConversation[] ) => {
+		const conversationsToCheck = conversations ? conversations : Smooch.getConversations();
+		const { unreadConversations, unreadMessages } = calculateUnread( conversationsToCheck );
+		setUnreadCount( unreadConversations );
+
+		return { unreadConversations, unreadMessages };
+	};
```

Comment:
Tangent: it seems like `setUnreadCount` is entirely not needed, and we can replace it with a resolver for `getUnreadCount`. This way everything will be automatically synced and cached.